lib/exabgp/reactor/api/command/text.py

```diff
@@ -32,22 +32,26 @@ def register (function):
 def _show_routes_callback(reactor, service, last, route_type, advertised, extensive):
 	def callback ():
 		families = None
+		lines_per_yield = environment.settings().api.chunk
 		if last in ('routes', 'extensive', 'static', 'flow', 'l2vpn'):
 			peers = reactor.peers.keys()
 		else:
 			peers = [n for n in reactor.peers.keys() if 'neighbor %s' % last in n]
+		line_count = 0
 		for key in peers:
 			peer = reactor.peers.get(key, None)
 			if not peer:
 				continue
 			if advertised:
 				families = peer._outgoing.proto.negotiated.families if peer._outgoing.proto else []
 			for change in list(peer.neighbor.rib.outgoing.sent_changes(families)):
```

Comment:
The pythonic way to do this would be to have something like
```
routes = list(peer.neighbor.rib.outgoing.sent_changes(families))
while routes:
   changes,routes = changes[: lines_per_yield],changes[lines_per_yield:]
   for change in changes:
      ....
   yield True
```
However performance wise I do not know how it compares.